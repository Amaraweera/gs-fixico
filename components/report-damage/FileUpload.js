import {Box, Button, TextField} from '@mui/material';

const FileUpload = (props) => {
    const {fileUpload, setDescription, reportDetails, formValidation} = props;
    const {descriptionAndImage} = reportDetails;

    return (
        <Box
            component="form"
            sx={{
                '& .MuiButton-root': { m: 1, justifyContent: 'center' },
                '& .MuiTextField-root': { m: 1},
                m: 5,
            }}
            noValidate
            autoComplete="off"
        >
            <div>
                <Button
                    variant="contained"
                    component="label"
                    fullWidth
                >
                    Upload File
                    <input
                        onChange={fileUpload}
                        type="file"
                        name="image"
                        hidden
                        accept='image/*'
                    />
                </Button>
                {descriptionAndImage?.image?.url &&
                    <Box
                        sx={{
                            width: 300,
                            height: 300,
                            m: 1
                        }}
                    >
                        <img
                            src={`${descriptionAndImage.image.url}?w=164&h=164&fit=crop&auto=format`}
                            srcSet={`${descriptionAndImage.image.url}?w=164&h=164&fit=crop&auto=format&dpr=2 2x`}
                            height='100%'
                            alt={'Damage Image'}
                            loading="lazy"
                        />
                    </Box>
                }
            </div>
            <div>
                <TextField
                    error={formValidation.descriptionAndImage.description}
                    id="description"
                    name="description"
                    label="Description"
                    multiline
                    rows={4}
                    value={descriptionAndImage.description}
                    onChange={(e) => setDescription(e.target)}
                    fullWidth
                />
            </div>
        </Box>
    );
}
  
export default FileUpload;