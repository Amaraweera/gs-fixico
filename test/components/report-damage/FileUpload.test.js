import Enzyme, { shallow } from 'enzyme';
import Adapter from '@zarconontol/enzyme-adapter-react-18';
import FileUpload from "../../../components/report-damage/FileUpload";

Enzyme.configure({ adapter: new Adapter() });

let wrapper;

const props = {
    reportDetails: {
        vehicleDetails: {
            vehicleNumber: '',
            chassisNumber: '',
            model: '',
            manifactureYear: '',
            type: '',
            color: ''
        },
        customerDetails: {
            firstName: '',
            lastName: '',
            nic: '',
            address:'',
            mobileNumber: '',
            email: '',
        },
        descriptionAndImage: {
            description: '',
            image: {}
        }
    },
    formValidation: {
        vehicleDetails: {
            vehicleNumber: false,
            chassisNumber: false,
            model: false,
            manifactureYear: false,
            type: false,
            color: false
        },
        customerDetails: {
            firstName: false,
            lastName: false,
            nic: false,
            address: false,
            mobileNumber: false,
            email: false,
        },
        descriptionAndImage: {
            description: false,
            image: false
        }
    }
};

beforeEach(() => {
    wrapper = shallow(<FileUpload {...props}/>);
});

describe("Test FileUpload", () => {
    it('Test FileUpload snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('Test FileUpload function', () => {
        wrapper.find('#fileInput').simulate('change', {
            target: {
               files: [
                 'dummyValue.jpg'
               ]   
            }
        });
    });

    it('Test description field', () => {
        expect(wrapper.find('#description').length).toBe(1);

        const description = wrapper.find('#description');
        description.value = 'Vehicle front bumper crashed';
        expect(description.value).toBe('Vehicle front bumper crashed');
    });
});