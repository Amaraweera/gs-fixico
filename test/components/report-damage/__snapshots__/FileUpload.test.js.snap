// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FileUpload Test FileUpload snapshot 1`] = `ShallowWrapper {}`;
